Tests to avoid duplicate results

ayush1298 opened the pull request:
Closes https://github.com/embeddings-benchmark/mteb/issues/3383

2 types of tests are added in this PR. 
1. test_duplicate_results.py: Finds all duplicate task results across revisions for each model. This uses the same logic that is used to identify duplicate tasks in [PR](https://github.com/embeddings-benchmark/results/pull/364). It will not delete any duplicate results, but tests will fail and will show these duplicate results. 
2. test_model_implementation.py: Checks that all models in the results folder are implemented in MTEB. If it fails, then it will return a list of unimplemented models.

ayush1298 committed:
Tests to avoid duplicate results

ayush1298 commented:
Currently, both of these tests are failing. The output are as follows(better for as a reference also, how it will look when tests fail): 


1. Test1
<details>
<summary>  tests/test_duplicate_results.py </summary>

```
tests/test_duplicate_results.py F                                [100%]

=============================== FAILURES ===============================
_________________ test_no_duplicate_task_results_exist _________________

duplicate_check = {'FacebookAI/xlm-roberta-base': [{'files': [{'file_path': PosixPath('/Users/ayushmunot/Desktop/OpenSource/results/resu...revision': '3', 'revisions_present': ['no_revision_available', '3'], 'task_name': 'MassiveIntentClassification', ...}]}

    def test_no_duplicate_task_results_exist(duplicate_check):
        """
        Test that no duplicate task results exist across revisions.
    
        A duplicate is when the same task exists in multiple revisions for the same model.
        These should be cleaned up by keeping only the reference revision or the first valid one.
        """
        if not duplicate_check:
            assert True
            return
    
        error_lines = [
            "Found duplicate task results that could be deleted:\n",
        ]
    
        for model_name, duplicates in sorted(duplicate_check.items()):
            reference_revision = get_reference_revision(model_name)
            error_lines.append(f"\n{model_name} (reference revision: {reference_revision}):")
    
            for dup in duplicates:
                error_lines.append(f"  Task: {dup['task_name']}")
                error_lines.append(f"    Revisions present: {dup['revisions_present']}")
                error_lines.append(f"    Retained revision: {dup['retained_revision']}")
                error_lines.append(f"    Should delete from revisions: {dup['to_delete']}")
    
        error_message = "\n".join(error_lines)
>       pytest.fail(error_message)
E       Failed: Found duplicate task results that could be deleted:
E       
E       
E       FacebookAI/xlm-roberta-base (reference revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089):
E         Task: AngryTweetsClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: BornholmBitextMining
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: DKHateClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: DalajClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: DanishPoliticalCommentsClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: LccSentimentClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: MassiveIntentClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: MassiveScenarioClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: NoRecClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: NordicLangClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E         Task: SweRecClassification
E           Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E           Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E           Should delete from revisions: ['no_revision_available']
E       
E       FacebookAI/xlm-roberta-large (reference revision: c23d21b0620b635a76227c604d44e43a9f0ee389):
E         Task: MassiveIntentClassification
E           Revisions present: ['no_revision_available', 'c23d21b0620b635a76227c604d44e43a9f0ee389']
E           Retained revision: c23d21b0620b635a76227c604d44e43a9f0ee389
E           Should delete from revisions: ['no_revision_available']
E         Task: MassiveScenarioClassification
E           Revisions present: ['no_revision_available', 'c23d21b0620b635a76227c604d44e43a9f0ee389']
E           Retained revision: c23d21b0620b635a76227c604d44e43a9f0ee389
E           Should delete from revisions: ['no_revision_available']
E       
E       openai/text-embedding-ada-002 (reference revision: 3):
E         Task: MassiveIntentClassification
E           Revisions present: ['no_revision_available', '3']
E           Retained revision: 3
E           Should delete from revisions: ['no_revision_available']

tests/test_duplicate_results.py:127: Failed
======================= short test summary info ========================
FAILED tests/test_duplicate_results.py::test_no_duplicate_task_results_exist - Failed: Found duplicate task results that could be deleted:
```

</details>

2.  Test2
<details>
<summary>  tests/test_model_implementation.py </summary>

```
tests/test_model_implementation.py F                             [100%]

=============================== FAILURES ===============================
_______________ test_all_models_are_implemented_in_mteb ________________

unimplemented_check = {'jinaai/jina-clip-v2': ['344d954da76eb8ad47a7aaff42d012e30c15b8fe'], 'sentence-transformers/multi-qa-mpnet-base-dot-v...c97fe237538cbe6e7f'], 'sentence-transformers/static-retrieval-mrl-en-v1': ['f60985c706f192d45d218078e49e5a8b6f15283a']}

    def test_all_models_are_implemented_in_mteb(unimplemented_check):
        """
        Test that all models in the results folder are implemented in MTEB.
    
        A model is considered implemented if:
        1. Its name appears in mteb.get_model_metas(), OR
        2. Its model_meta.json file contains a name that appears in mteb.get_model_metas()
        """
        if not unimplemented_check:
            assert True
            return
    
        error_lines = [
            "Found models in results that are not implemented in MTEB:\n",
        ]
    
        for model_name, revisions in sorted(unimplemented_check.items()):
            error_lines.append(f"  {model_name}")
            error_lines.append(f"    Revisions: {revisions}")
    
        error_lines.append(
            "\nThese models need to be added to MTEB's model implementations "
            "or the results should be removed."
        )
    
        error_message = "\n".join(error_lines)
>       pytest.fail(error_message)
E       Failed: Found models in results that are not implemented in MTEB:
E       
E         jinaai/jina-clip-v2
E           Revisions: ['344d954da76eb8ad47a7aaff42d012e30c15b8fe']
E         sentence-transformers/multi-qa-mpnet-base-dot-v1
E           Revisions: ['3af7c6da5b3e1bea796ef6c97fe237538cbe6e7f']
E         sentence-transformers/static-retrieval-mrl-en-v1
E           Revisions: ['f60985c706f192d45d218078e49e5a8b6f15283a']
E       
E       These models need to be added to MTEB's model implementations or the results should be removed.

tests/test_model_implementation.py:98: Failed
======================= short test summary info ========================
FAILED tests/test_model_implementation.py::test_all_models_are_implemented_in_mteb - Failed: Found models in results that are not implemented in MTEB:
```

</details>


Should I also add script/code to remove duplicate results?

ayush1298 committed:
fix tests

Samoed commented on tests/test_duplicate_results.py:
```diff
@@ -0,0 +1,124 @@
+        model_meta = mteb.get_model_meta(model_name)
+        return model_meta.revision
+    except Exception:
+        return None
```
I think it's better to fail if there is exception
```suggestion
```

ayush1298 replied:
If there is no model on HF, it will call from_hub(), but for proprietary models, these will end up with HFValidationError, so what should do in that case.  Below is error when I have removed these try-except: 

<details>
<summary> Error trace </summary>

```
tests/test_duplicate_results.py E                                [100%]

================================ ERRORS ================================
________ ERROR at setup of test_no_duplicate_task_results_exist ________

    @pytest.fixture
    def duplicate_check():
        """Fixture to find duplicates."""
>       return find_duplicate_tasks()

tests/test_duplicate_results.py:94: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_duplicate_results.py:29: in find_duplicate_tasks
    reference_revision = get_reference_revision(model_name)
tests/test_duplicate_results.py:10: in get_reference_revision
    model_meta = mteb.get_model_meta(model_name)
../mteb/mteb/models/get_model_meta.py:132: in get_model_meta
    meta = ModelMeta.from_hub(model_name, revision)
../mteb/mteb/models/model_meta.py:412: in from_hub
    meta = cls._from_hub(model, revision, compute_metadata)
../mteb/mteb/models/model_meta.py:308: in _from_hub
    if model_name and compute_metadata and repo_exists(model_name):
../../../miniconda3/lib/python3.12/site-packages/huggingface_hub/utils/_validators.py:106: in _inner_fn
    validate_repo_id(arg_value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

repo_id = 'Cohere/Cohere-embed-v4.0_(output_dtype=binary)'

    def validate_repo_id(repo_id: str) -> None:
        """Validate `repo_id` is valid.
    
        This is not meant to replace the proper validation made on the Hub but rather to
        avoid local inconsistencies whenever possible (example: passing `repo_type` in the
        `repo_id` is forbidden).
    
        Rules:
        - Between 1 and 96 characters.
        - Either "repo_name" or "namespace/repo_name"
        - [a-zA-Z0-9] or "-", "_", "."
        - "--" and ".." are forbidden
    
        Valid: `"foo"`, `"foo/bar"`, `"123"`, `"Foo-BAR_foo.bar123"`
    
        Not valid: `"datasets/foo/bar"`, `".repo_id"`, `"foo--bar"`, `"foo.git"`
    
        Example:
        ```py
        >>> from huggingface_hub.utils import validate_repo_id
        >>> validate_repo_id(repo_id="valid_repo_id")
        >>> validate_repo_id(repo_id="other..repo..id")
        huggingface_hub.utils._validators.HFValidationError: Cannot have -- or .. in repo_id: 'other..repo..id'.
        ```
    
        Discussed in https://github.com/huggingface/huggingface_hub/issues/1008.
        In moon-landing (internal repository):
        - https://github.com/huggingface/moon-landing/blob/main/server/lib/Names.ts#L27
        - https://github.com/huggingface/moon-landing/blob/main/server/views/components/NewRepoForm/NewRepoForm.svelte#L138
        """
        if not isinstance(repo_id, str):
            # Typically, a Path is not a repo_id
            raise HFValidationError(f"Repo id must be a string, not {type(repo_id)}: '{repo_id}'.")
    
        if repo_id.count("/") > 1:
            raise HFValidationError(
                "Repo id must be in the form 'repo_name' or 'namespace/repo_name':"
                f" '{repo_id}'. Use `repo_type` argument if needed."
            )
    
        if not REPO_ID_REGEX.match(repo_id):
>           raise HFValidationError(
                "Repo id must use alphanumeric chars or '-', '_', '.', '--' and '..' are"
                " forbidden, '-' and '.' cannot start or end the name, max length is 96:"
                f" '{repo_id}'."
            )
E           huggingface_hub.errors.HFValidationError: Repo id must use alphanumeric chars or '-', '_', '.', '--' and '..' are forbidden, '-' and '.' cannot start or end the name, max length is 96: 'Cohere/Cohere-embed-v4.0_(output_dtype=binary)'.

../../../miniconda3/lib/python3.12/site-packages/huggingface_hub/utils/_validators.py:160: HFValidationError
======================= short test summary info ========================
ERROR tests/test_duplicate_results.py::test_no_duplicate_task_results_exist - huggingface_hub.errors.HFValidationError: Repo id must use alphanum...
=========================== 1 error in 4.13s ===========================
```

</details>

Samoed replied:
It seems that for cohere models it's saved a bit different, than in mteb, because in mteb it implemented with name `Cohere/Cohere-embed-v4.0 (output_dtype=binary)` (with space before dtype)

ayush1298 replied:
got it. So, should we rename it? It should be same models, right

Samoed replied:
No, in mteb we're creating folder with https://github.com/embeddings-benchmark/mteb/blob/42dea0150cb270f0d9485c6d4b40ea678391259d/mteb/cache.py#L82

Probably you can rely on values from model meta

ayush1298 replied:
I am still not getting, how to avoid these. https://github.com/embeddings-benchmark/results/pull/383#discussion_r2649138877

Samoed replied:
I've updated a bit function for getting reference revision. You can change it if you want

Samoed commented on tests/test_duplicate_results.py:
```diff
@@ -0,0 +1,124 @@
+        return None
+
+
+def find_duplicate_tasks() -> Dict[str, List[Dict]]:
```
Use types for 3.10+ (just dict and list)

Samoed commented on tests/test_duplicate_results.py:
```diff
@@ -0,0 +1,124 @@
+    duplicates_found = {}
+
+    for model_folder in sorted(results_folder.glob("*")):
+        if not model_folder.is_dir() or model_folder.name.startswith(".") or model_folder.name == "Human":
```
Can you run ruff format? (Maybe we need add linter to format)

ayush1298 replied:
Its changing many files, better to run `ruff format` in different PR and then merge it.

Samoed replied:
You can run only for this file

ayush1298 replied:
Okay, should I also add the linter also in Makefile?

Samoed replied:
Not, for now

Samoed commented on tests/test_duplicate_results.py:
```diff
@@ -0,0 +1,124 @@
+@pytest.fixture
+def duplicate_check():
+    """Fixture to find duplicates."""
+    return find_duplicate_tasks()
```
```suggestion
```

Samoed commented on tests/test_duplicate_results.py:
```diff
@@ -0,0 +1,124 @@
+                    "revision": revision,
+                    "file_path": task_file,
+                    "revision_folder": revision_folder,
+                })
```
I think this would be simpler
```suggestion
                task_to_revisions[task_name].append(revision)
```

Samoed commented on tests/test_duplicate_results.py:
```diff
@@ -0,0 +1,124 @@
+        # Find tasks with duplicates
+        model_duplicates = []
+        for task_name, revision_list in sorted(task_to_revisions.items()):
+            if len(revision_list) > 1:
```
```suggestion
            if len(revision_list) == 1:
                continue
```

Samoed commented on tests/test_model_implementation.py:
```diff
@@ -0,0 +1,95 @@
+@pytest.fixture
+def unimplemented_check():
+    """Fixture to find unimplemented models."""
+    return find_unimplemented_models()
```
```suggestion
```

Samoed commented on tests/test_model_implementation.py:
```diff
@@ -0,0 +1,95 @@
+                        impl_exists = True
+                        break
+            except (json.JSONDecodeError, IOError) as e:
+                pytest.fail(f"Error reading {meta_file}: {e}")
```
I think it's easier to not catch exception

Samoed commented on tests/test_model_implementation.py:
```diff
@@ -0,0 +1,95 @@
+        implementations = {meta.name for meta in mteb.get_model_metas() if meta.name}
+        return implementations
+    except Exception as e:
+        pytest.skip(f"Could not load MTEB model implementations: {e}")
```
There won't be error and easier to use in code
```suggestion
```

ayush1298 committed:
address comments and ruff format

ayush1298 committed:
address comments

Samoed committed:
add fallback for reference revision

ayush1298 commented:
1. jinaai/jina-clip-v2 (https://huggingface.co/jinaai/jina-clip-v2)
2. sentence-transformers/multi-qa-mpnet-base-dot-v1 (https://huggingface.co/sentence-transformers/multi-qa-mpnet-base-dot-v1)
3. sentence-transformers/static-retrieval-mrl-en-v1 (https://huggingface.co/sentence-transformers/static-retrieval-mrl-en-v1)

for these 3 models, we dont have implementations. All of these are on hf. Should I create issue for them?

Samoed committed:
raise if not found

Samoed committed:
add comment

Samoed commented:
> for these 3 models, we dont have implementations. All of these are on hf. Should I create issue for them?

We have issues for them 
- https://github.com/embeddings-benchmark/mteb/issues/3725
- https://github.com/embeddings-benchmark/mteb/issues/3724

ayush1298 commented:
Should I remove those duplicate results also?

Samoed commented:
Yes

ayush1298 committed:
update script and remove duplicate results

ayush1298 commented:
Below is a trace when test_duplicate_results was failing. I have removed them now, and also added model which are not implemented in exclude_models in this test, so tests would not fail because of them. 


<details>
<summary> Test trace </summary>

```
E           Failed: Found duplicate task results that could be deleted:
E           
E           
E           FacebookAI/xlm-roberta-base:
E             Task: AngryTweetsClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: BornholmBitextMining
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: DKHateClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: DalajClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: DanishPoliticalCommentsClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: LccSentimentClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: MassiveIntentClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: MassiveScenarioClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: NoRecClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: NordicLangClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E             Task: SweRecClassification
E               Revisions present: ['no_revision_available', 'e73636d4f797dec63c3081bb6ed5c7b0bb3f2089']
E               Retained revision: e73636d4f797dec63c3081bb6ed5c7b0bb3f2089
E               Should delete from revisions: ['no_revision_available']
E           
E           FacebookAI/xlm-roberta-large:
E             Task: MassiveIntentClassification
E               Revisions present: ['no_revision_available', 'c23d21b0620b635a76227c604d44e43a9f0ee389']
E               Retained revision: c23d21b0620b635a76227c604d44e43a9f0ee389
E               Should delete from revisions: ['no_revision_available']
E             Task: MassiveScenarioClassification
E               Revisions present: ['no_revision_available', 'c23d21b0620b635a76227c604d44e43a9f0ee389']
E               Retained revision: c23d21b0620b635a76227c604d44e43a9f0ee389
E               Should delete from revisions: ['no_revision_available']
E           
E           openai/text-embedding-ada-002:
E             Task: MassiveIntentClassification
E               Revisions present: ['no_revision_available', '3']
E               Retained revision: 3
E               Should delete from revisions: ['no_revision_available']

tests/test_duplicate_results.py:156: Failed
```

</details>

ayush1298 committed:
skipping deleted files in test

ayush1298 committed:
exclude unimplemented models to pass tests

ayush1298 commented:
Removed duplicate results. Also, excluded `"jinaai/jina-clip-v2", "sentence-transformers/multi-qa-mpnet-base-dot-v1", "sentence-transformers/static-retrieval-mrl-en-v1"` from tests, so tests will pass now. In the future, when implementation for these models is added, we will remove it.

Samoed commented on tests/test_no_splits_are_removed.py:
```diff
@@ -32,7 +32,7 @@ def test_no_splits_are_removed():
     for filepath_str in changed_files:
         filepath = root / Path(filepath_str)
-        if filepath.name == "model_meta.json":
+        if not filepath.exists() or filepath.name == "model_meta.json":
```
I think you shouldn't add this, because test should fail on your changes

ayush1298 replied:
Won't it also fail when I am deleting files?

Samoed replied:
Wouldn't this make `no_splits_are_removed` less useful?

ayush1298 replied:
But it would not check it based on revision, like if for a model, the same tasks results are present in 2 different revisions, so we can delete the file from 1 revision and its not loss of split, but though this test will fail in that case as It will just check if deleted files causing some splits to be lost, though they will not as it will be present in some other revision of a model.

Samoed replied:
Let's keep it as is then

ayush1298 commented:
@KennethEnevoldsen Could you check this one also?

ayush1298 committed:
remove exclude models

ayush1298 committed:
Merge branch 'main' into test_for_duplicate_results

ayush1298 commented:
@Samoed Did it take time to update once models is added in MTEB, because tests are still failing as they were not able to find model. I have also merge main branch.

Samoed commented:
Yes. Mteb wasn't released after model merge. You can wait until next release or you can try to install from repository directly

ayush1298 commented:
> Yes. Mteb wasn't released after model merge. You can wait until next release or you can try to install from repository directly

For tests to pass, I have to wait, right? btw, where I have to install from repository.

Samoed commented:
It's set pyproject, but not sure if we should change this https://github.com/embeddings-benchmark/results/blob/feaf749ed52fba2b232320bea4302c93a6206a49/pyproject.toml#L7

ayush1298 commented:
> It's set pyproject, but not sure if we should change this
> 
> https://github.com/embeddings-benchmark/results/blob/feaf749ed52fba2b232320bea4302c93a6206a49/pyproject.toml#L7

I think, its better to wait.

Samoed commented:
New mteb version with added missing models released. Now test is failing, because recently `bflhc/Octen-Embedding-8B` added new revision of their model

ayush1298 commented:
> New mteb version with added missing models released. Now test is failing, because recently `bflhc/Octen-Embedding-8B` added new revision of their model

there are 2 revisions: 
'f7db178d5a82fb841f606a6a67c423cead2fdbba' - new revision,
 '2030603c2926ab005fafd824fac5911e271be21f' - old revision
 
out of which and tests is suggesting to delete results of '2030603c2926ab005fafd824fac5911e271be21f', which is old one. So, should I delete that?

Samoed commented:
I'm not sure. I don't think we should require for users delete old revisions to pass tests

ayush1298 commented:
> I'm not sure. I don't think we should require for users delete old revisions to pass tests

Isn't the aim to not again populate the results repo with duplicate results? Or else we can remove from these as tests, and make these as a script which we will run in a few days using Github Actions?
@KennethEnevoldsen What's your thought on this one?

KennethEnevoldsen commented:
Hmm I think the goal would be to avoid duplicates going forward and if a user submits a PR with duplicates (e.g. for the `bflhc/Octen-Embedding-8B` case) then I would like them or us to be prompted to consider whether existing results should be deleted. I think adding a test would ensure that - that would in some case require users to delete outdated results.

Let me know what you guys think

ayush1298 commented:
> Hmm I think the goal would be to avoid duplicates going forward and if a user submits a PR with duplicates (e.g. for the `bflhc/Octen-Embedding-8B` case) then I would like them or us to be prompted to consider whether existing results should be deleted. I think adding a test would ensure that - that would in some case require users to delete outdated results.
> 
> Let me know what you guys think

I also think same. I think we should keep tests, and user can decide which results they want to keep. In test, we can suggest which tasks have duplicate revision which we are doing now, and then they can decide what to keep and what to delete.

ayush1298 commented:
@KennethEnevoldsen Should I delete the duplicate results then for `bflhc/Octen-Embedding-8B` and then we can finalize this PR?

KennethEnevoldsen commented:
please do!

ayush1298 committed:
remove duplicate results for bflhc/Octen-Embedding-8B

ayush1298 commented:
> please do!

Removed duplicate results of `bflhc/Octen-Embedding-8B`

KennethEnevoldsen reviewed (approved):
@Samoed what are your thoughts here?

ayush1298 committed:
Merge remote-tracking branch 'origin/main' into test_for_duplicate_results

ayush1298 committed:
remove duplicate results after merging main

Samoed reviewed (approved):
I think this is ready to merge

ayush1298 commented:
Could we merge it then, if there is no other blockage?
